src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.cs

```diff
@@ -20,5 +24,55 @@ internal static void VerifyValidPath(string path, string argName)
                 throw new ArgumentException(SR.Argument_InvalidPathChars, argName);
             }
         }
+
+        internal static bool Copy(string sourcePath, string destinationPath, bool recursive,
+            bool skipExistingFiles, CancellationToken cancellationToken)
+        {
+            var searchOption = recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+            IEnumerable<string> directoryEnumeration;
+
+            try
+            {
+                directoryEnumeration = Directory.EnumerateDirectories(sourcePath, "*", searchOption);
+            }
+            catch (IOException)
+            {
+                // Return false if Enumeration is not possible
+                return false;
```

Comment:
Shouldn't this throw?

Reply:
As per API review it should not throw if it fails to read/enumerate but I cannot remember what the answer was whether it should just continue with next file and still return false in the end. I believe it should and then this implementation needs to be changed.

Reply:
@adamsitnik Could you confirm if the part is a proper implementation?

Reply:
@bartonjs Could you say something about this while its still in your memory?

Reply:
I believe that the feeling "in the room" was that the copy would do as much as it could; so this would be more appropriate as `ret = false;` than `return false;` since this is skipping later, potentially completeable, work.

Reply:
Updated it, @bartonjs is this fine so for you?